src/components/article-body/body-components/codeblock.tsx

```diff
@@ -7,18 +8,68 @@ import {
 } from '../../../utils/get-code-height';
 
 export const CodeBlock = ({ lang, value }: { lang: string; value: string }) => {
+    // This is a workaround to a bug where a long line of code will be cut off at the end.
+    // This seems to be a CodeMirror issue, but is especially apparent here without line wrapping.
+    // https://github.com/codemirror/codemirror5/issues/5639 describes a similar issue.
+
+    // We're utilizing the Intersection Observer API (https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)
+    // to force the component to rerender when the Code Block elemtn comes into the viewport.
+
+    const [, forceUpdate] = useReducer(x => x + 1, 0); // https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate
+    const wrapper = useRef<HTMLDivElement>(null);
+
+    const onVisible = useCallback(
+        (
+            entries: IntersectionObserverEntry[],
+            observer: IntersectionObserver
+        ) => {
+            // When this gets called with the element intersecting the vieport, we rerender and disconnect the observer.
+            // We are only ever registering 1 element to the observer, so we can disconnnect whenever this gets called with an entry.
+            if (entries.length && entries[0].isIntersecting) {
+                forceUpdate();
+                observer.disconnect();
+            }
+        },
+        []
+    );
+
+    useEffect(() => {
+        let observer: IntersectionObserver;
+        if (wrapper.current) {
+            // Set up the observer to execute the onVisible callback when 1% of the block is in the viewport.
+            observer = new IntersectionObserver(onVisible, {
+                threshold: 0.01,
+            });
+            observer.observe(wrapper.current);
+        }
+
+        return () => {
+            // Clean up and disconnect, needed if the user hasn't scrolled the code block into view.
+            if (observer) {
+                observer.disconnect();
+            }
+        };
+    }, [wrapper]);
+
```

Comment:
@justinmckenzie @max-melo So I used IntersectionObserver along with the `forceUpdate` logic to just rerender once when the element is scrolled into view. Seems like the simplest  workaround to this issue at the moment, but let me know what you think.

Reply:
is this when the code block is large ?

Reply:
No this can happen on both the `large` and `auto` setting.